src/Ocelot.Provider.Consul/ConsulProviderFactory.cs

```diff
@@ -28,10 +28,11 @@ private static IServiceDiscoveryProvider CreateProvider(IServiceProvider provide
         var factory = provider.GetService<IOcelotLoggerFactory>();
         var consulFactory = provider.GetService<IConsulClientFactory>();
         var contextAccessor = provider.GetService<IHttpContextAccessor>();
+        var context = contextAccessor.HttpContext;
 
         var configuration = new ConsulRegistryConfiguration(config.Scheme, config.Host, config.Port, route.ServiceName, config.Token); // TODO Why not to pass 2 args only: config, route? LoL
-        contextAccessor.HttpContext.Items[nameof(ConsulRegistryConfiguration)] = configuration; // initialize data
-        var serviceBuilder = provider.GetService<IConsulServiceBuilder>(); // consume data in default/custom builder
+        context.Items[nameof(ConsulRegistryConfiguration)] = configuration; // initialize data
+        var serviceBuilder = context.RequestServices.GetService<IConsulServiceBuilder>(); // consume data in default/custom builder
```

Comment:
AFAIK ASP.NET creates a DI scope for `HttpContext` of each HTTP request (see [this link](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-3.0#request-services-1) for more info).
Technically, the solutions are equivalent, but I thought that since we already access `HttpContext` via `IHttpContextAccessor`, it would be logical to access the scope of the HTTP request instead of creating a new one.